lib/datadog/tracing/distributed/skip_policy.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Tracing
+    module Distributed
+      # Helper method to decide when to skip distributed tracing
+      module SkipPolicy
+        module_function
+
+        # Skips distributed tracing if disabled for this instrumentation
+        # or if APM is disabled unless there is an AppSec event (from upstream distributed trace or local)
+        #
+        # pin_config is a Datadog::Core::Pin object, which gives the configuration of a single instance of an object
+        # global_config is the config for all instances of a framework
+        def skip?(pin_config: nil, global_config: nil, trace: nil)
+          if ::Datadog.configuration.appsec.standalone.enabled &&
+              (trace.nil? || trace.get_tag(::Datadog::AppSec::Ext::TAG_DISTRIBUTED_APPSEC_EVENT) != '1')
```

Comment:
```suggestion
          if (trace.nil? || trace.get_tag(::Datadog::AppSec::Ext::TAG_DISTRIBUTED_APPSEC_EVENT) != '1') &&
            ::Datadog.configuration.appsec.standalone.enabled
```
I recommend inverting the check order since the `trace` checks are much cheaper to do than the `configuration` look up.

Reply:
Fixed in 2eac1e8f668717d1904b18e49b43690723546a95 !